packages/flutter_tools/lib/src/vscode/vscode.dart

```diff
@@ -178,3 +185,12 @@ class VsCode {
     return json['version'];
   }
 }
+
+class _VsCodeInstallLocation {
+  final String installPath;
+  final String extensionsFolder;
+  final String edition;
+  final bool isInsiders;
+  _VsCodeInstallLocation(this.installPath, this.extensionsFolder, { this.edition, bool isInsiders })
```

Comment:
Nit: this can be `const`

Reply:
Oh yeah, done! (though some calls to it still use `new` because they use an env var).